src/main/java/org/team1540/robot2024/subsystems/vision/AprilTagVision.java

```diff
@@ -0,0 +1,84 @@
+package org.team1540.robot2024.subsystems.vision;
+
+import edu.wpi.first.math.geometry.Pose2d;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import org.littletonrobotics.junction.Logger;
+import org.team1540.robot2024.Constants;
+import org.team1540.robot2024.util.vision.TimestampedVisionPose;
+
+import java.util.Arrays;
+import java.util.function.Consumer;
+
+public class AprilTagVision extends SubsystemBase {
+    private final AprilTagVisionIO frontCameraIO;
+    private final AprilTagVisionIOInputsAutoLogged frontCameraInputs = new AprilTagVisionIOInputsAutoLogged();
+
+    private final AprilTagVisionIO rearCameraIO;
+    private final AprilTagVisionIOInputsAutoLogged rearCameraInputs = new AprilTagVisionIOInputsAutoLogged();
+
+    private final Consumer<TimestampedVisionPose> visionPoseConsumer;
+
+    private TimestampedVisionPose frontPose =
+            new TimestampedVisionPose(-1, new Pose2d(), new int[0], new Pose2d[0]);
+    private TimestampedVisionPose rearPose =
+            new TimestampedVisionPose(-1, new Pose2d(), new int[0], new Pose2d[0]);
+
+    public AprilTagVision(
+            AprilTagVisionIO frontCameraIO,
+            AprilTagVisionIO rearCameraIO,
+            Consumer<TimestampedVisionPose> visionPoseConsumer) {
+        this.frontCameraIO = frontCameraIO;
+        this.rearCameraIO = rearCameraIO;
+        this.visionPoseConsumer = visionPoseConsumer;
+    }
+
+    @Override
+    public void periodic() {
+        frontCameraIO.updateInputs(frontCameraInputs);
+        rearCameraIO.updateInputs(rearCameraInputs);
+        Logger.processInputs("Vision/FrontCamera", frontCameraInputs);
+        Logger.processInputs("Vision/RearCamera", rearCameraInputs);
+
+        if (frontCameraInputs.lastMeasurementTimestampSecs > frontPose.timestampSecs()) {
+            frontPose = new TimestampedVisionPose(
+                    frontCameraInputs.lastMeasurementTimestampSecs,
+                    frontCameraInputs.estimatedPoseMeters,
+                    frontCameraInputs.seenTagIDs,
+                    frontCameraInputs.tagPosesMeters);
+        }
+        if (rearCameraInputs.lastMeasurementTimestampSecs > rearPose.timestampSecs()) {
+            rearPose = new TimestampedVisionPose(
+                    rearCameraInputs.lastMeasurementTimestampSecs,
+                    rearCameraInputs.estimatedPoseMeters,
+                    rearCameraInputs.seenTagIDs,
+                    rearPose.tagPosesMeters());
```

Comment:
Why are you continually passing an empty array of poses to these guys?

Reply:
Uh I shouldn't be? These values should be updated in `updateInputs`.